misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -133,6 +130,45 @@ internal class RealTransacter private constructor(
     }
   }
 
+  override fun <T> failSafeRead(block: (session: Session) -> T): T {
+    check(!inTransaction) {
+      "Cann't do fail safe read inside a transaction"
+    }
+
+    return maybeWithTracing(APPLICATION_TRANSACTION_SPAN_NAME) {
+      transactionWithRetriesInternal {
+        maybeWithTracing(DB_TRANSACTION_SPAN_NAME) {
+          failSafeWithoutTransactionInternalSession(block)
+        }
+      }
+    }
+  }
+
+  private fun <T> failSafeWithoutTransactionInternalSession(block: (session: RealSession) -> T): T {
+    check(config.type.isVitess) {
+      "Only applicable with vitess connection"
+    }
+
+    return withSession { session ->
+      try {
+        session.target(Destination(TabletType.MASTER)) {
+          block(session)
+        }
+      } catch (e: Throwable) {
```

Comment:
Never catch Throwable: https://rules.sonarsource.com/java/tag/error-handling/RSPEC-1181

Catch SQLException instead. Can we find out exactly SQLException is being thrown? Is it an SQLRecoverableException for example? We should be as specific as possible so we don't hide other errors.